contracts/FreakTeam.sol

```diff
@@ -0,0 +1,297 @@
+    event deletedFreak(address _address, uint16 employeeNumber);
+    event promotedFreak(uint16 employeeNumber, Skill oldSkillLevel, Skill newSkillLevel);
+    event changedFreakNorm(uint16 employeeNumber, Norm oldNorm, Norm newNorm);
+
+    constructor(address hr, address financial) ERC1155("") {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(HR_ROLE, hr);
+        _setupRole(FINANCIAL_ROLE, financial);
+        _setRoleAdmin(FREAK_ROLE, HR_ROLE);
+    }
+
+    modifier onlyHR() {
+        require(hasRole(HR_ROLE, msg.sender), "Caller is not a hr");
+        _;
+    }
+
+    /**
+    @notice Add a new Freak
+    @param  _address  Freak wallet address
+    @param _startDate Date of employment
+    @param _employeeNumber  Employee number
+    @param _role  The role he owns
+    @param _skill The skill he owns
+    @param _norm The norm he owns
+     */
+    function addNewFreak(
+        address _address,
+        string memory _name,
+        uint256 _startDate,
+        uint16 _employeeNumber,
+        Role _role,
+        Skill _skill,
+        Norm _norm
+    ) external onlyHR {
+        require(freaks[_address].employeeNumber == 0, "On the address exists already one freak");
+        grantRole(FREAK_ROLE, _address);
+        freaks[_address] = Freak(
+            _name,
+            _startDate,
+            0,
+            _employeeNumber,
+            _role,
+            _skill,
+            _norm,
+            freakScore(_role, _skill, _norm, _employeeNumber)
+        );
+
+        freakAccounts.push(_address);
+        _mint(_address, _employeeNumber, 1, "");
+        emit addedFreak(
+            _address,
+            _name,
+            _startDate,
+            _employeeNumber,
+            _role,
+            _skill,
+            _norm,
+            freakScore(_role, _skill, _norm, _employeeNumber)
+        );
+    }
+
+    /**
+        @notice Delete an employee from AgileFreaks
+        @param _address Freak address
+     */
+    function deleteFreak(address _address) external onlyHR {
+        require(_address != msg.sender, "Caller can not remove himself");
+        require(
+            balanceOf(_address, freaks[_address].employeeNumber) > 0,
+            "Address should contain at least 1 token"
+        );
+        freaks[_address].stopDate = block.timestamp;
+        _burn(_address, freaks[_address].employeeNumber, 1);
+        emit deletedFreak(_address, freaks[_address].employeeNumber);
+    }
+
+    /**
+        @notice Promote one freak by changing his skill
+        @param _address Freak address
+        @param newSkill The skill obtained after promote
+     */
+    function promoteFreak(address _address, Skill newSkill) external onlyHR {
+        require(_address != msg.sender, "Caller can not promote himself");
+        Skill oldSkill = freaks[_address].skill;
+        freaks[_address].skill = newSkill;
+        emit promotedFreak(freaks[_address].employeeNumber, oldSkill, newSkill);
+    }
```

Comment:
We need to check if the new skill is better to be sure that we really promote the Freak.
require(
            _newSkill > freaks[_address].skill
        );